Readme.md

```diff
@@ -1,5 +1,262 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+Before we start to define anything for the lexer you will need to learn the syntax for [BNF](#bnf)
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set up some token culling rules
+
+Here is an example of what our BNF rules look like
```

Comment:
Be a little more specific - Here is a simple BNF example, defining a space-delimited list of integers

Reply:
What do you think of something like this - Here is a simple BNF example, that is defining a space, a number and a list of numbers